packages/mui-material/src/Popover/Popover.js

```diff
@@ -376,60 +405,52 @@ const Popover = React.forwardRef(function Popover(inProps, ref) {
   const container =
     containerProp || (anchorEl ? ownerDocument(resolveAnchorEl(anchorEl)).body : undefined);
 
-  const externalForwardedProps = {
-    slots,
-    slotProps: {
-      ...slotProps,
-      paper: externalPaperSlotProps,
+  const [RootSlot, { slots: rootSlots, slotProps: rootSlotPropsProp, ...rootProps }] = useSlot(
+    'root',
+    {
+      ref,
+      elementType: PopoverRoot,
+      externalForwardedProps: {
+        ...externalForwardedProps,
+        ...other,
+      },
+      shouldForwardComponentProp: true,
+      additionalProps: {
+        slots: { backdrop: slots.backdrop },
+        slotProps: { backdrop: mergeSlotProps(slotProps.backdrop, { invisible: true }) },
+        container,
+        open,
+      },
+      ownerState,
+      className: clsx(classes.root, className),
     },
-  };
+  );
 
   const [PaperSlot, paperProps] = useSlot('paper', {
+    ref: paperRef,
+    className: classes.paper,
     elementType: PopoverPaper,
     externalForwardedProps,
+    shouldForwardComponentProp: true,
     additionalProps: {
       elevation,
-      className: clsx(classes.paper, externalPaperSlotProps?.className),
-      style: isPositioned
-        ? externalPaperSlotProps.style
-        : { ...externalPaperSlotProps.style, opacity: 0 },
+      style: isPositioned ? undefined : { opacity: 0 },
     },
     ownerState,
   });
 
-  const [RootSlot, { slotProps: rootSlotPropsProp, ...rootProps }] = useSlot('root', {
-    elementType: PopoverRoot,
-    externalForwardedProps,
-    additionalProps: {
-      slotProps: { backdrop: { invisible: true } },
-      container,
-      open,
-    },
-    ownerState,
-    className: clsx(classes.root, className),
-  });
-
-  const handlePaperRef = useForkRef(paperRef, paperProps.ref);
-
   return (
     <RootSlot
       {...rootProps}
-      {...(!isHostComponent(RootSlot) && { slotProps: rootSlotPropsProp, disableScrollLock })}
-      {...other}
-      ref={ref}
+      {...(!isHostComponent(RootSlot) && {
+        slots: rootSlots,
+        slotProps: rootSlotPropsProp,
+        disableScrollLock,
+      })}
     >
-      <TransitionComponent
-        appear
-        in={open}
-        onEntering={handleEntering}
-        onExited={handleExited}
-        timeout={transitionDuration}
-        {...TransitionProps}
-      >
-        <PaperSlot {...paperProps} ref={handlePaperRef}>
-          {children}
-        </PaperSlot>
-      </TransitionComponent>
+      <TransitionSlot {...transitionSlotProps} timeout={transitionDuration}>
```

Comment:
Should we move the `timeout` prop to `useSlot` as well?

Reply:
I can't because of line `399`, the `transitionDuration` depends on the TransitionSlot.